gravitee-am-identityprovider/gravitee-am-identityprovider-common/src/main/java/io/gravitee/am/identityprovider/common/oauth2/authentication/AbstractOpenIDConnectAuthenticationProvider.java

```diff
@@ -179,67 +181,66 @@ private UriBuilder prepareSignInUrl(String redirectUri) {
             builder.addParameter(io.gravitee.am.common.oidc.Parameters.NONCE, SecureRandomString.generate());
         }
 
+        // using response_mode parameter with the default value for selected resposne_type is not recommended
```

Comment:
* regarding the suggestion: it should have been "==", not "!=" (the idea is that we include the `response_mode` in the sign in url only if it's different than the default for the selected `response_type`)
* as for the question: I feel like 99.9% of cases, users should be fine with `default`. If not, the most likely case is that they're working around some bug or "uniqueness" of the upstream provider, because the AM is what handles he response, and we can handle both types just fine. This leads me to think it's better to leave a possiblity of "unsafe" config to get broader compatiblity. It might be worth displaying some kind of warning in the UI though.